spec/app/domain/authentication/validate_security_spec.rb

```diff
@@ -251,4 +256,57 @@ def resource_class(returned_resource)
       expect { subject }.to raise_error(Errors::Authentication::Security::AccountNotDefined)
     end
   end
+
+  context "with the default role class" do
+    let(:admin_roleid) { [test_account, 'user', 'admin'].join(':') }
+    let(:admin_role_double) { double('admin_role_double') }
+
+    let(:ws_name) { 'service1' }
+    let(:ws_id) { ['conjur', fake_authenticator_name, ws_name].join('/') }
+    let(:ws_resourceid) { [test_account, 'webservice', ws_id].join(':') }
+    let(:ws_double) { double('ws_resource_double') }
+    
+    let(:user_id) { 'some-user' }
+    let(:user_roleid) { [test_account, 'user', user_id].join(':') }
+    let(:user_role_double) { double('user_role_double') }
+
+    before do 
+      allow(Role).to receive(:[]).with(admin_roleid).and_return(admin_role_double)
+      allow(Role).to receive(:roleid_from_username).and_return(user_roleid)
+      
+      allow(Resource).to receive(:[]).with(ws_resourceid).and_return(ws_double)
+
+      allow(user_role_double).to receive(:allowed_to?).with('authenticate', ws_double).and_return(true)
+    end
```

Comment:
We shouldn't have references to `Role` or `Resource` in these tests.  If this is necessary, it means that we have hardcoded deps that shouldn't be there.  I tried to look to find what was going on, but....

see my comment below....

Reply:
I think to test this properly, we need to stub out database access only, and not anything above that. That's why there are currently references to these two classes.

To remove them, we'll need to ask the subject what its `role_class` and `webservice_resource_class` are and stub out methods on those, instead.

That seems a little weird to me, but only a little. WDYT?

Reply:
Actually, looks like this is already taken care of in the specs. I'll update them and remove these references.

Reply:
So `ValidateSecurity` (and all the CommandClasses) were designed specifically to make this kind of unit testing easy:

```ruby
    ValidateSecurity = CommandClass.new(
      dependencies: {
        role_class: ::Authentication::MemoizedRole,
        webservice_resource_class: ::Resource,
        logger: Rails.logger
      },
```
Simply inject the doubles you need for `role_class` and `webservice_resource_class`.  Currently the subject is being instantiated with default deps, ie, the ones we want in prod.  We'll just be using mocks here instead.